### PR TITLE
Removes maps from basemaps

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Did you know that a map being in basemap doesn't even do anything?
I removed all maps and everything (barring central command, which is the only thing in ``DEFAULT_MAP_TRAITS`` (the 'hardcoded' map list that has to be loaded through basemap) ) worked perfectly fine, including reading map config (icebox's captain getting a banner for example), map rotation and selection (which uses maps.txt config), and modular maps (tram's maints). Because I also removed templates.dm (a path that DOESN'T EXIST), I also checked that ruins also spawn in, just to be safe.

I first found out about this after I heard a downstream was running maps without the maps in the basemap, because they forgot to add it, but it worked anyways.

## Why It's Good For The Game

What's the point of this code? It doesn't seem to do anything at all, so instead of confusing people, we should just remove it.

## Changelog

Nothing player-facing.

### CODE REVIEW

File: code/__DEFINES/maps.dm [hunk 1]
```diff
@@ -3,8 +3,8 @@ The /tg/ codebase allows mixing of hardcoded and dynamically-loaded Z-levels.
 Z-levels can be reordered as desired and their properties are set by "traits".
 See code/datums/map_config.dm for how a particular station's traits may be chosen.
 The list DEFAULT_MAP_TRAITS at the bottom of this file should correspond to
-the maps that are hardcoded, as set in _maps/_basemap.dm. SSmapping is
-responsible for loading every non-hardcoded Z-level.
+the maps that are hardcoded, generally Central Command.
```
Comment (san7890) [inline code/__DEFINES/maps.dm:6]:
```suggestion
the maps that are hardcoded, generally Central Command (centcom.dmm).
```

just to tie the concept of central command to the actual map file

**Comment by msgerbs** [comment]
Can we get a comment somewhere clarifying what basemap.dm is actually for? After this PR the only reference to the file is a comment further down saying that "DEFAULT_MAP_TRAITS" must correspond to basemap.dm which... Honestly doesn't really explain anything at all.

**Comment by JohnFulpWillard** [comment]
I'll look into the rabbit hole later but for now I'll just say ``ALL_MAPS`` define doesn't exist and I've been trying to look at https://github.com/tgstation/tgstation/pull/34571 https://github.com/tgstation/tgstation/pull/35339 https://github.com/tgstation/tgstation/pull/60404 and https://github.com/tgstation/tgstation/pull/35339 - the solution is probably in one of these but I can't look into it at the moment

**Comment by Mothblocks** [comment]
its used in unit tests ya dingus

https://github.com/tgstation/tgstation/blob/master/.github/workflows/ci_suite.yml#L84

**Comment by Mothblocks** [comment]
what did u think "compile maps" was :P

**Comment by Mothblocks** [comment]
> Because I also removed templates.dm (a path that DOESN'T EXIST), 

Because it's made in CI

https://github.com/tgstation/tgstation/blob/d56b76210398b5a7f40dee40660ad200ae74748f/tools/build/build.js#L50-L65

**Comment by Mothblocks** [comment]
https://fs.blog/chestertons-fence/

**Comment by Mothblocks** [comment]
that being said i would be ok if you wanted to add a comment documenting this. kind of scared of what wouldve happened if i didnt see this pr in bed

**Comment by msgerbs** [comment]
So we have a code file that exists only to create a define that is only used for a CI test which references a file only created during CI? Surely there's a better way to make that CI test work.

**Comment by Mothblocks** [comment]
What sets compile maps apart is that we can perform it very quickly, over every single map, and never run a server at any point. The templates includes every single ruin. We cannot run all of these at the same time on CI, nor do we want to. I really don't think there's anything wrong with the setup other than that it's obviously not documented well

**Pull request closed without merging**